tokenizer.go

```diff
@@ -67,6 +88,142 @@ func FromFile(path string) (*Tokenizer, error) {
 	return &Tokenizer{tokenizer: tokenizer}, nil
 }
 
+// FromPretrained downloads necessary files and initializes the tokenizer.
+// Parameters:
+//   - modelID: The Hugging Face model identifier (e.g., "bert-base-uncased").
+//   - destination: Optional. If provided and not nil, files will be downloaded to this folder.
+//     If nil, a temporary directory will be used.
+//   - authToken: Optional. If provided and not nil, it will be used to authenticate requests.
+func FromPretrained(modelID string, destination, authToken *string) (*Tokenizer, error) {
```

Comment:
let's add a test to exercise default and optional params, e.g. with and without optional auth token